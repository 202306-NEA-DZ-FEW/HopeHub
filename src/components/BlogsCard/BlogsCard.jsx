import React from "react";
import Link from "next/link";
import Image from "next/image";
import { useTranslation } from "next-i18next";

const BlogCard = ({ image, title, subtitle, author, blogId, summary }) => {
    const { t } = useTranslation("common");

    return (
        <div className='flex flex-col lg:flex-row'>
            <Link href={`/blogs/${blogId}`}>
                <div className='image-container'>
                    <Image
                        src={image}
                        alt={title}
                        width={300}
                        height={200}
                        className='image'
                    />
                </div>
            </Link>
            <div className='flex flex-col my-2 lg:px-6 text-poppins'>
                <Link href={`/blogs/${blogId}`}>
                    <h2 className='text-sm lg:text-xl font-semibold'>
                        {title}
                    </h2>
                </Link>
                <p className='text-gray-500 font-normal text-xs lg:text-sm'>
                    {t("Written by")} {author}
                </p>
                <p className='text-gray-600 text-xs lg:text-base font-semibold mt-2'>
                    {subtitle}
                </p>
                <p className='text-gray-600 text-xs lg:text-base'>{summary}</p>
            </div>
        </div>
    );
};

export default BlogCard;
