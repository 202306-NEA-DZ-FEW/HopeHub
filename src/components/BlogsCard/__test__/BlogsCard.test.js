import renderer from "react-test-renderer";
import BlogsCard from "../BlogsCard";

it("renders correctly", () => {
    const tree = renderer.create(<BlogsCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
