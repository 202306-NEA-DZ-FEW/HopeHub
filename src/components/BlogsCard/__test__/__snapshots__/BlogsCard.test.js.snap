// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col lg:flex-row"
>
  <a
    href="/blogs/undefined"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="image-container"
    >
      <img
        className="image"
        data-nimg="1"
        decoding="async"
        height={200}
        onError={[Function]}
        onLoad={[Function]}
        src=""
        style={
          Object {
            "color": "transparent",
          }
        }
        width={300}
      />
    </div>
  </a>
  <div
    className="flex flex-col my-2 lg:px-6 text-poppins"
  >
    <a
      href="/blogs/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <h2
        className="text-sm lg:text-xl font-semibold text-NeutralBlack dark:text-NeutralWhite"
      />
    </a>
    <p
      className="text-gray-500 dark:text-gray-400 font-light text-xs lg:text-sm"
    >
      Written by 
    </p>
    <p
      className="text-NeutralBlack dark:text-NeutralWhite font-light text-xs lg:text-base py-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </div>
</div>
`;
